### DISCUSSION TITLE
How to set fold method for neovim tree sitter?

### DISCUSSION THREAD

**Discussion by sp71** [LazyVim/LazyVim#1233] [Q&A] [answered]
Tree Sitter offers code folding. In which file would we set the configuration to use tree sitter folding? Unsure where to place this [configuration](https://github.com/nvim-treesitter/nvim-treesitter#folding) in LazyVim. What are people's thoughts about setting this by default?

**Comment by amaanq** [comment, marked as answer]
```lua
vim.opt.foldmethod = "expr"
vim.opt.foldexpr = "nvim_treesitter#foldexpr()"
```

in your nvim config at `lua/config/options.lua`

**Reply by sp71** [reply]
How would you also set the last command?

`set nofoldenable                     " Disable folding at startup.`

Tried `vim.opt.nofoldenable = false` but still everytime a file is opened, everything is collapsed by default

**Reply by amaanq** [reply]
not sure, maybe try `vim.cmd([[ set nofoldenable]])` but that's not ideal. I don't see why everything would collapse when you open a file, that doesn't happen for me

**Reply by sp71** [reply]
That seems to be the default setting without the nofoldenable flag. This [article](https://www.jmaguire.tech/posts/treesitter_folding/) corroborates this behavior. Your command did the trick ✅

**Reply by sp71** [reply]
Set Folding in `lua/config/options.lua`

```
vim.opt.foldmethod = "expr"
vim.opt.foldexpr = "nvim_treesitter#foldexpr()"
vim.cmd([[ set nofoldenable]])
```

**Reply by dpetka2001** [reply]
I believe it should be `vim.opt.foldenable = false`. Not `nofoldenable`. That's just for setting it up with Vimscript I think, like you do with `vim.cmd`.

**Reply by javier-elias** [reply]
for me only work with:
set foldmethod=indent
